project11/project11.sql

```diff
@@ -0,0 +1,248 @@
+	country TEXT NOT NULL UNIQUE,
+    -- В одном объекте может жить только одна делегация
+	house_id INT NOT NULL UNIQUE,
+	name TEXT,
+    -- У делегации только один руководитель
+	leader_id INT NOT NULL UNIQUE,
+	leader_name TEXT NOT NULL,
+    -- Один телефон принадлежит только одному руководителю
+	leader_telephone TEXT NOT NULL UNIQUE,
+    
+    -- Других ключей нет
+
+    -- Связь 1:1 между делегациями и объектами. В одном объекте может располагаться только одна делегация и наоборот.
+	FOREIGN KEY(house_id) REFERENCES object_(id)
+);
+
+-- Служебная таблица для пола
+CREATE TYPE sex AS ENUM('Мужской', 'Женский', 'Custom');
+
+-- Карта регистрации спортсмена определяет остальные его атрибуты (TEXT, так как может быть leading zero)
+-- имя спортсмена и его опциональные атрибуты (возраст, вес, рост и т.д.)
+-- Внешние ссылки на id делегации и id помещения (inf <-> 1 в обоих случаях)
+-- Сейчас может возникнуть неконсисентность, что house_id оказался не id жилого дома.
+CREATE TABLE sportsman(
+    card_id VARCHAR(5) NOT NULL,
+    name TEXT NOT NULL, 
+    age SMALLINT NOT NULL, 
+    weight SMALLINT NOT NULL, 
+    height SMALLINT NOT NULL, 
+    sex SEX NOT NULL,
+
+	delegation_id INT NOT NULL, 
+    house_id INT NOT NULL,
+	
+	-- Ключи
+    --card_id может принадлежать только одному спортсмену
+	PRIMARY KEY(card_id),
+	-- Спортсмен может жить только в одном объекте.
+	UNIQUE(card_id, house_id),
+    -- Каждый спортсмен может быть только в одной делегации
+    UNIQUE(card_id, delegation_id),
+    -- Других ключей нет
+	
+	-- Связь N:1 между спортсмена и делегациями. Каждый спортсмен может быть только в одной делегации. В каждой делегации может быть много спортсменов.
+	FOREIGN KEY(delegation_id) REFERENCES delegation(id),
+    -- Связь N:1 между спортсменами и объектами. Каждый спортсмен может быть в одном объекте. В каждом объекте может быть много спортсменов.
+	FOREIGN KEY(house_id) REFERENCES object_(id),
+    -- Проверка на возраст, вес и рост.
+	CHECK (age > 0), 
+    CHECK (weight > 0), 
+    CHECK (height > 0)
+);
+
+
+--------------------------------------------- SPORTS ----------------------------------------------------------------
+-- Служебная табличка с id видов спорта
+CREATE TYPE sport AS ENUM('Фехтование', 'Баскетбол', 'Плавание', 'Футбол');
+
+-- Отношение между спортсменами и
+-- видами спорта в которых они участвуют
+-- отношение inf <-> inf
+CREATE TABLE SportsmanToSport(
+    sportsman_id TEXT NOT NULL, 
+    sport SPORT NOT NULL,
+
+    -- Спортсмен не может дважды участвовать в одном виде спорта.
+	UNIQUE(sportsman_id, sport),
+	-- Других ключей нет
+
+    -- Связь N:M между спортсменами и видами спорта. Один спортсмен может участвовать в многих видах спорта. Одним видом спорта могут заниматься много спортсменов.
+	FOREIGN KEY(sportsman_id) REFERENCES sportsman(card_id)
+);
+
+-- Отношение между объектами и
+-- видами спорта для которых они предназначены
+-- отношение inf <-> inf
+CREATE TABLE object_ToSport(
```

Comment:
So do you prefer CamelCase or snake_case after all?)

Reply:
_diversity!_

Reply:
done